compiler/src/iree/compiler/InputConversion/StableHLO/Preprocessing/StableHLOToStableHLO.cpp

```diff
@@ -1540,6 +1540,71 @@ struct CustomCallIsTopK final
   }
 };
 
+struct IotaSortSliceIsTopK final : OpRewritePattern<mlir::stablehlo::SortOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mlir::stablehlo::SortOp op,
+                                PatternRewriter &rewriter) const override {
+    auto opOperands = op.getOperands();
+    auto opResults = op.getResults();
+    Value topKInput;
+    if (opOperands.size() != 2 || opResults.size() != 2) {
+      return rewriter.notifyMatchFailure(
+          op, "Slice that maps to TopK must have exactly two inputs/outputs");
+    }
+
+    bool isInputIota = false;
```

Comment:
Rather than a booleans, you could set this to be a `Value iotaOperand` and set to the appropriate operand.

Reply:
I'm not entirely sure what you mean by that - are you suggesting to get rid of the for-loop altogether and just use operands[1]?
Otherwise, I don't see a value in creating a separate iotaOperand since it's not used later on (unless you just mean setting it only if the cast succeeds, which at that point it's just a boolean with extra steps)

Reply:
Ah no. Just instead of doing `bool isInputIota = false;` you can do `Value inputIota;`, then rather than setting true / false, you can set it to the corresponding `Value`. Then after the check you can do `if (!inputIota) return failures();`. Its just nice as complexity is the same, but you get to have the actual value.

You get more opportunity to tweak things in the future without rewriting the checkers.

Reply:
Done, thanks for the clarification